### PR TITLE
fix: removing dependancy for node-requests

### PR DESCRIPTION
**Pull request by manchuck**
This PR removes the 'requests' package, which has not been updated in over 6 years and has major security errors.

### CODE REVIEW

**Review by jeffswartz — COMMENTED** [review]
Some tests in opentok_spec.js are failing.

**Review by jeffswartz — CHANGES_REQUESTED** [review]
https://github.com/opentok/opentok-node/pull/311#pullrequestreview-1662523959

**Comment by manchuck** [comment]
> Some tests are failing: [#311 (review)](https://github.com/opentok/opentok-node/pull/311#pullrequestreview-1662523959).
> 
> Also, please rev the version (to 2.17.0) in package.json and package-lock.json.

This should only be requested once the PR is ready to merge in. Otherwise the PR will always be rejected as we paly whack-a-mole with the release versions

**Comment by manchuck** [comment]
> Some tests in opentok_spec.js are failing.

These failures are inconsistent and intermittent. I am not familiar with these tests or how to resolve them. The reason for asking for your review on this PR is because there is not full coverage for all use cases with the SDK. Any guidance from you would be helpful

**Review by jeffswartz — CHANGES_REQUESTED** [review]
15 tests are failing ("Archiving listArchives should return an array of archives and a total count" and 14 others).

**Review by jeffswartz — COMMENTED** [review]
When I install packages, I am alerted the `get-func-name` is audited with "high" vulnerability. And `@babel/traverse` is flagged as being critical.  Can we update chai and nyc to update these sub-packages?

**Review by jeffswartz — CHANGES_REQUESTED** [review]
We need to rev the version of the library (to 2.16.1) in the package.json and package-lock.json files.

File: lib/archiving.js [hunk 1]
```diff
@@ -191,13 +191,12 @@ function Archive(config, properties) {
...
   var rurl = config.apiEndpoint + '/v2/project/' + config.apiKey + path;
-  request.defaults(_.pick(config, 'proxy', 'timeout'))(
-    {
-      url: rurl,
-      method: method,
-      headers: generateHeaders(config),
-      json: body
-    },
+  config.headers = generateHeaders(config);
+  request.defaults(_.pick(config, 'proxy', 'timeout'));
```
Comment (jeffswartz) [inline lib/archiving.js:195]: @manchuck The failing tests all have this message:

```
Error: Nock: No match for request {
      "method": "post",
      "url": "https://api.opentok.com/v2/project/APIKEY/archive",
      "headers": {
        "user-agent": "OpenTok-Node-SDK/2.16.0",
        "x-opentok-auth": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJBUElLRVkiLCJpc3QiOiJwcm9qZWN0IiwiaWF0IjoxNjk3NjcwOTM3LCJleHAiOjE2OTc2NzEyMzd9.2Ja75MtjsYP4cLvt-uYfXfUhIs07np0SLc9CbuZ2yUY",
        "accept": "application/json",
        "content-length": 190,
        "content-type": "application/x-www-form-urlencoded",
        "host": "api.opentok.com"
      },
      "body": "sessionId=1_MX4xMDB-MTI3LjAuMC4xflR1ZSBKYW4gMjggMTU6NDg6NDAgUFNUIDIwMTR-MC43NjAyOTYyfg&name=test_archive_name&hasAudio=&hasVideo=&outputMode=&layout=&resolution=&streamMode=&multiArchiveTag="
    }
```
Do we need to change something in these lines?

File: lib/archiving.js [hunk 2]
```diff
@@ -287,26 +305,32 @@ exports.startArchive = function (ot, config, sessionId, options, callback) {
     function startArchiveCallback(err, response, body) {
       if (err) {
         callback(err);
+        return;
       }
-      else if (response.statusCode !== 200) {
-        if (response && response.statusCode === 404) {
+      try {
+        body = JSON.parse(body);
+      } catch {}
```
Comment (jeffswartz) [inline lib/archiving.js:312]: The linter complains that this (and other instances of `catch {}`) results in a parsing error. (Replace with `catch (error) { }`). 

There are already existing linting errors in the main branch. But if you add `/* eslint-env es2018 */` to the top of this file, you will see more linting errors.

**Review by jeffswartz — APPROVED** [review]

**Review by jeffswartz — CHANGES_REQUESTED** [review]
Please rev the version (to 2.17.0) in package.json and package-lock.json. (I assume we are ready to release this.)

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**